samcli/lib/telemetry/metric.py

```diff
@@ -148,16 +148,18 @@ def wrapped(*args, **kwargs):
             # Execute the function and capture return value. This is returned back by the wrapper
             # First argument of all commands should be the Context
             return_value = func(*args, **kwargs)
-        except UserException as ex:
+        except (UserException, click.BadOptionUsage, click.BadArgumentUsage, click.BadParameter) as ex:
```

Comment:
why did we add more exceptions to catch?

Reply:
I think I understand now, but why did we add only these? Should we just catch all click exceptions? Or at least a few others that are here: https://click.palletsprojects.com/en/8.1.x/api/#exceptions

Reply:
Good catch. I missed `click.UsageError` and probably `click.NoSuchOption`. 

The idea of grouping these click exceptions together is that these exceptions should be fixable by users. Any user-fixable-exception that is thrown within the call stacks wrapped by `@track_command` should be caught by this line. I don't want to catch `click.ClickException` in general because that could be over-catching. I'd rather an exception to be an Unhandled Exception by default.

Reply:
Gotcha. What about `NoSuchOption`?

Reply:
I don't see any reason we should use `NoSuchOption` in our own code. So don't think we should include it.

Reply:
Maybe I'm misunderstanding but wouldn't `NoSuchOption` be raised by `click` when a user provides an option that doesn't exist? Or are you saying that wouldn't even reach this code anyway?

Reply:
It won't reach this code. We should have all `@option` declared before `@track_command`